### DISCUSSION TITLE
OpenApi Generator

### DISCUSSION THREAD

**Discussion by daniel-sc** [mikecao/flight#576] [General]
I created a (simple) OpenApi generator for Flight: https://github.com/OpenAPITools/openapi-generator/pull/18406

Feedback and improvements welcome :)

**Comment by n0nag0n** [comment]
Hey there! This is awesome! I made a couple comments on the PR.

**Reply by daniel-sc** [reply]
@n0nag0n would it make sense to document this on the website or should i rather make a separate blog post?

**Reply by n0nag0n** [reply]
So I actually would love some docs on how you set this up, or a blog post about it. I'm happy to stick it right in the documentation, or take excepts from a blog, whatever would work easiest for you. I was actually hit up about this earlier this week and I had to find the old PR you had done! @daniel-sc

**Reply by daniel-sc** [reply]
@n0nag0n ok, published a first version: https://dev.to/danielsc/define-generate-and-implement-an-api-first-approach-with-openapi-generator-and-flightphp-1fb3 please share any feedback, happy to improve :)

**Reply by n0nag0n** [reply]
So awesome! I'm going to put this in the official docs with some information about it! Thank you!

**Reply by n0nag0n** [reply]
Made a little blurb [here](https://docs.flightphp.com/awesome-plugins/awesome-plugins#api-documentation) and [here](https://docs.flightphp.com/media)

**Reply by daniel-sc** [reply]
@n0nag0n thanks! One minor correction - you wrote

>  Blog post written by Daniel Schreiber on how to use the OpenAPI Generator with FlightPHP to generate API documentation.

I think this is misleading, as this is not about generation of "API documentation" but the opposite way - you manually define your API documentation (or it is given) and generate php flight stubs + models from it?

**Reply by n0nag0n** [reply]
Ok, wow it was late and I just accepted what Copilot did. Shame on me. :) But I did have some more time to go through the blog post and I think there's a couple things to clear up.

I think there's supposed to be a composer require in there somewhere? In the blog post you mention `RegisterRoutes::registerRoutes()` but I can't quite figure out where that line of code came from. I don't understand how the UserApi class registers an actual route for `Flight::route()` or if that's still a manual step. Also one other quick thing since this is an API, I've created a new method called `Flight::jsonHalt()` which is exactly like halt, but in JSON format.

**Reply by daniel-sc** [reply]
@n0nag0n thanks for the feedback! I updated the blog post - would you mind checking again, if it is (more) understandable now?

**Reply by n0nag0n** [reply]
Ahh yes, that makes more sense now, thank you!
